patch/llpcPatchDescriptorLoad.cpp

```diff
@@ -78,6 +78,84 @@ PatchDescriptorLoad::PatchDescriptorLoad()
     initializePatchDescriptorLoadPass(*PassRegistry::getPassRegistry());
 }
 
+// =====================================================================================================================
+/// Create a non-strided, non-swizzled buffer descriptor.
+///
+/// @param baseAddr (i64) base address of the descriptor (top 16 bits are ignored);
+///                 for convenience, an <N x i32> is also accepted (using the low 48 bits)
+/// @param size (i32) size of the referenced buffer in bytes
+Value* PatchDescriptorLoad::buildBufferDescriptor(Value* baseAddr,
+                                                  Value* size,
+                                                  Instruction* insertPt)
+{
+    IRBuilder<> builder(insertPt);
+    Value* bufDesc = UndefValue::get(m_pContext->Int32x4Ty());
+
+    Value* elem0;
+    Value* elem1;
+
+    if (baseAddr->getType() == m_pContext->Int64Ty()) {
+        elem0 = builder.CreateTrunc(baseAddr, m_pContext->Int32Ty());
+        elem1 = builder.CreateLShr(baseAddr, 32, "", true);
+        elem1 = builder.CreateTrunc(elem1, m_pContext->Int32Ty());
+    } else {
+        elem0 = builder.CreateExtractElement(baseAddr, (uint64_t)0);
+        elem1 = builder.CreateExtractElement(baseAddr, (uint64_t)1);
```

Comment:
I don't think (uint64_t)1 is necessary. The argument (1) will be converted to u64 unambiguously.

(uint64_t)1 -> 1

If we really want a i64 literal, 1ull is enough.